data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java

```diff
@@ -238,38 +255,53 @@ public static ConnectionStringBuilder createWithAadApplicationCertificateSubject
             throw new IllegalArgumentException("privateKey cannot be null");
         }
 
-        ConnectionStringBuilder csb = new ConnectionStringBuilder(resourceUri);
+        ConnectionStringBuilder csb = new ConnectionStringBuilder(clusterUrl);
         csb.applicationClientId = applicationClientId;
         csb.x509CertificateChain = x509CertificateChain;
         csb.privateKey = privateKey;
         csb.aadAuthorityId = authorityId;
         return csb;
     }
 
-    public static ConnectionStringBuilder createWithAadAccessTokenAuthentication(String resourceUri, String token) {
-        if (StringUtils.isEmpty(resourceUri)) {
-            throw new IllegalArgumentException("resourceUri cannot be null or empty");
+    public static ConnectionStringBuilder createWithAadAccessTokenAuthentication(String clusterUrl, String token) {
+        if (StringUtils.isEmpty(clusterUrl)) {
+            throw new IllegalArgumentException("clusterUrl cannot be null or empty");
         }
         if (StringUtils.isEmpty(token)) {
             throw new IllegalArgumentException("token cannot be null or empty");
         }
 
-        ConnectionStringBuilder csb = new ConnectionStringBuilder(resourceUri);
+        ConnectionStringBuilder csb = new ConnectionStringBuilder(clusterUrl);
         csb.accessToken = token;
         return csb;
     }
 
-    public static ConnectionStringBuilder createWithAadTokenProviderAuthentication(String resourceUri, Callable<String> tokenProviderCallable) {
-        if (StringUtils.isEmpty(resourceUri)) {
-            throw new IllegalArgumentException("resourceUri cannot be null or empty");
+    public static ConnectionStringBuilder createWithAadTokenProviderAuthentication(String clusterUrl, Callable<String> tokenProviderCallable) {
+        if (StringUtils.isEmpty(clusterUrl)) {
+            throw new IllegalArgumentException("clusterUrl cannot be null or empty");
         }
 
         if (tokenProviderCallable == null) {
             throw new IllegalArgumentException("tokenProviderCallback cannot be null");
         }
 
-        ConnectionStringBuilder csb = new ConnectionStringBuilder(resourceUri);
+        ConnectionStringBuilder csb = new ConnectionStringBuilder(clusterUrl);
         csb.tokenProvider = tokenProviderCallable;
         return csb;
     }
-}
+
+    public static ConnectionStringBuilder createWithAadManagedIdentity(String clusterUrl) {
```

Comment:
This re-order is confusing.
Is there any actual change to this class other than adding this method and the two members?

Reply:
Yeah, sorry. Need to make reviewers suffer once when we organize it, and then it should make it easier to read in the future.

The changes are:
1) These 2 methods that I added
2) Introducing an approach to CSB that makes more sense - an explicit "use" boolean to determine which flow is desired, instead of the TokenFactory trying to guess based on which properties are set. This actually became necessary for the first time now, and I'll likely want to expand the concept in the future.